addons/hr_payroll/static/src/js/benefit.js

```diff
@@ -12,9 +12,84 @@ odoo.define('hr_payroll.benefit.view_custo', function(require) {
     var BenefitCalendarController = CalendarController.extend({
 
         events: {
-            'click .btn-benefit-generate': '_onGenerateBenefits',
-            'click .btn-benefit-validate': '_onValidateBenefits',
             'click .btn-payslip-generate': '_onGeneratePayslips',
+            'click .btn-benefits-conflict': '_onBenefitsConflict',
+        },
+
+        /**
+         * @override
+         */
+        reload: function () {
+            var date_fmt = 'YYYY-MM-DD HH:mm:ss';
+            var self = this;
+            this.firstDay = this.model.data.target_date.clone().startOf('month');
+            this.lastDay = this.model.data.target_date.clone().endOf('month');
+
+            var date = new Date();
+            this.end_current_month = new Date(date.getFullYear(), date.getMonth() + 1, 0);
+
+            var super_function = this._super;
+            var args = arguments;
+
+            if (this.firstDay < this.end_current_month)
+            {
+                return this._rpc({
+                    model: 'hr.employee',
+                    method: 'generate_benefit',
+                    args: [this.firstDay.format(date_fmt), this.lastDay.format(date_fmt)],
+                }).then(function () {
+                    self.events = self._checkDataInRange(self.firstDay, self.lastDay, self.model.data.data);
+                    return self._rpc({
+                        model: 'hr.benefit',
+                        method: 'action_validate',
+                        args: [_.map(self.events, function (event) { return event.record.id; })],
+                    });
+                }).then(function() {
+                    return super_function.apply(self, args);
+                })
+            }
+            else
+            {
+                return super_function.apply(self, args);
+            }
+        },
+
+        /**
+         * @override
+         */
+        start: function () {
```

Comment:
Try to extract the date manipulation + rpc code to a specific method and reuse it in the ``start`` and the ``reload``